_posts/2017-10-09-functional-calisthenics.md

```diff
@@ -0,0 +1,123 @@
+---
+author: Jorge Gueorguiev Garcia
+comments: true
+date: 2017-10-09 13:00:00 +00:00
+layout: post
+asset-type: post
+slug: functional-calisthenics
+title: Functional Calisthenics
+tags:
+- functional programming
+---
+# The Setup
+
+While working on a Clojure application that was for production, rather than being as an exercise, I saw that I was using OO programming with the syntax of Clojure. I was having problems coming back to change code that I have done not long before. Testing was an absolute pain, with continuous use of with-redefs and weird subnesting of let statements happening all over the place. Clearly, my knowledge, and my internalization of functional programming, wasn't good enough. It was time to start looking into how to improve.
+
+Thanks to my colleagues at Codurance I discovered that at Socrate 2015 there was a discussion already about Functional Calisthenics, organized by Ian Johnson, and which results you can find at his blog post<a href="http://blog.ninjaferret.co.uk/2015/06/05/Introducing-Functional-Calisthenics.html">Introducing Functional Calisthenics</a>.
```

Comment:
`at Socrate 2015` is missing an `s`. Maybe it would be interesting to specify which country as well.